### PR TITLE
Simplify handle.zig

### PR DESCRIPTION
**Pull request by garettbass**

### CODE REVIEW

File: libs/zpool/src/handle.zig [hunk 1]
```diff
@@ -57,78 +57,71 @@ pub fn Handle(
...
+            var u = CompactUnion{ .id = self.id };
             return .{
-                .index = self.index(),
-                .cycle = self.cycle(),
+                .index = u.bits.index,
+                .cycle = u.bits.cycle,
             };
         }
 
-        pub fn index(self: Self) AddressableIndex {
```
Comment (garettbass) [inline libs/zpool/src/handle.zig:104]: Favor operating on `AddressableHandle` in all cases where `index` and `cycle` are needed.  The goal is an API that discourages shifting and masking the `index` and `cycle` bits for every operation.

File: libs/zpool/src/handle.zig [hunk 2]
```diff
@@ -57,78 +57,71 @@ pub fn Handle(
...
-            pub fn compact(handle: AddressableHandle) Compact {
-                const _index = @intCast(Index, handle.index);
-                const _cycle = @intCast(Cycle, handle.cycle);
-                return Compact.init(_index, _cycle);
-            }
-
-            pub fn eql(a: AddressableHandle, b: AddressableHandle) bool {
-                return a.index == b.index and a.cycle == b.cycle;
+            /// Returns the corresponding `Handle`
+            pub fn compact(handle: AddressableHandle) CompactHandle {
```
Comment (garettbass) [inline libs/zpool/src/handle.zig:121]: Perhaps I should rename this method simply `handle()`

Comment (michal-z) [reply libs/zpool/src/handle.zig:121]: Yes, I think we should go with handle().

**Comment by michal-z** [comment]
Looks good! Can you please rebase to latest?

**Comment by garettbass** [comment]
I have rebased and changed `AddressableHandle.compact()` to `AddressableHandle.handle()` as discussed.

**Pull request merged**